File: src/algorithm/RotateArray.java
package algorithm;

public class RotateArray {
	public static void main(String[] args) {
		int idx = 1;
		int[][] arr = new int[7][7];

		int start = 0;
		int end = arr.length;
		while (true) {
			for (int i = start; i < end; i++) {
				arr[start][i] = idx++;
			}
			for (int i = start + 1; i < end; i++) {
				arr[i][end - 1] = idx++;
			}
			for (int i = end - 2; i >= start; i--) {
				arr[end - 1][i] = idx++;
			}
			for (int i = end - 2; i > start; i--) {
				arr[i][start] = idx++;
			}
			start++;
			end--;
			if (start >= end)
				break;
		}

		for (int i = 0; i < arr.length; i++) {
			for (int j = 0; j < arr[i].length; j++) {
				System.out.printf("%3d ", arr[i][j]);
			}
			System.out.println();
		}

	}
}

File: src/algorithm/RotateArray.py
def rotate_array():
    idx = 1
    n = 7
    arr = [[0] * n for _ in range(n)]
    start = 0
    end = n

    while True:
        for i in range(start, end):
            arr[start][i] = idx
            idx += 1
        for i in range(start + 1, end):
            arr[i][end - 1] = idx
            idx += 1
        for i in range(end - 2, start - 1, -1):
            arr[end - 1][i] = idx
            idx += 1
        for i in range(end - 2, start, -1):
            arr[i][start] = idx
            idx += 1

        start += 1
        end -= 1
        if start >= end:
            break

    for row in arr:
        for element in row:
            print(f"{element:3d}", end=" ")
        print()


rotate_array()
